kyo-core/shared/src/test/scala/kyoTest/FlatTest.scala

```diff
@@ -61,6 +61,20 @@ class FlatTest extends KyoTest:
             succeed
         }
 
+        "effect mismatch" in {
+            def test[T](v: T < Fibers)(using Flat[T < Fibers]): T < Fibers = v
+            test(1)
+            test(1: Int < Fibers)
+            assertDoesNotCompile("test(1: Int < Options)")
+        }
+
+        "flat flat" in pendingUntilFixed {
+            def test[T](v: T < Fibers)(using Flat[T]): T < Fibers = v
+            test(1)
+            test(1: Int < Fibers)
+            assertDoesNotCompile("test(1: Int < Options)")
```

Comment:
This is currently compiling but shouldn't. It seems a simple workaround would be migrating all flat checks to the format with pending effects (`Flat[T < Fibers]` instead of `Flat[T]`) but it'd be ideal to fix this bug to reduce the noise of the flat check in Kyo's codebase.